x/crosschain/keeper/evm_deposit.go

```diff
@@ -36,11 +42,17 @@ func (k Keeper) HandleEVMDeposit(ctx sdk.Context, cctx *types.CrossChainTx) (boo
 	}
 
 	if inboundCoinType == coin.CoinType_Zeta {
-		// if coin type is Zeta, this is a deposit ZETA to zEVM cctx.
-		err := k.fungibleKeeper.DepositCoinZeta(ctx, to, inboundAmount)
+		indexBytes, err := cctx.GetCCTXIndexBytes()
 		if err != nil {
 			return false, err
 		}
+		// if coin type is Zeta, this is a deposit ZETA to zEVM cctx.
+		evmTxResponse, err := k.fungibleKeeper.ZEVMDepositAndCallContract(ctx, sender, to, inboundSenderChainID, inboundAmount, data, indexBytes)
```

Comment:
```suggestion
		evmTxResponse, err := k.fungibleKeeper.ZETADepositAndCallContract(ctx, sender, to, inboundSenderChainID, inboundAmount, data, indexBytes)
```

Would be more consistent with `ZRC20DepositAndCallContract` to specify the token IMO